_studio/shared/umc/codec/h265_dec/src/umc_h265_segment_decoder_dxva.cpp

```diff
@@ -135,6 +135,24 @@ void TaskBrokerSingleThreadDXVA::Start()
 
     TaskBroker_H265::Start();
     m_completedQueue.clear();
+
+    H265_DXVA_SegmentDecoder * dxva_sd = static_cast<H265_DXVA_SegmentDecoder *>(m_pTaskSupplier->m_pSegmentDecoder[0]);
+
+    if (dxva_sd == nullptr || dxva_sd->m_va == nullptr || dxva_sd->m_va->IsUseStatusReport())
```

Comment:
1. Having `dxva_sd == nullptr || dxva_sd->m_va == nullptr` is logic error, I prefer to throw exception here.
2. Is see that u try to minimize code changes, but for me, it would be better not to spread completion logic over several functions and do all things at one place - `TaskBrokerSingleThreadDXVA::GetNextTaskInternal`